.github/workflows/automerge.yml

```diff
@@ -0,0 +1,48 @@
+name: automerge
+on:
+  pull_request:
+    types:
+      - labeled
+      - unlabeled
+      - synchronize
+      - opened
+      - edited
+      - ready_for_review
+      - reopened
+      - unlocked
+  pull_request_review:
+    types:
+      - submitted
+  check_suite:
+    types:
+      - completed
+  status: {}
+jobs:
+  automerge:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: Generate token
+      id: generate_token
+      env:
+        ID: ${{secrets.APP_ID}}
+        PK: ${{secrets.APP_PRIVATE_KEY}}
+      run: |
+        issueTime=$(date +%s)
+        expireTime=$(date -d "$expireTime + 600 seconds" +%s)
+        header=$(echo '{ "alg": "RS256", "typ": "JWT" }' | jq -r '(. | @base64)')
+        payload=$(echo '{"iss": '"$ID"',"iat": '$issueTime' ,"exp": '$expireTime'}'| jq -r '(. | @base64)'| sed s/\+/-/ | sed -E s/=+$//)
+        echo "$PK" > key
+        signature=$(echo -n "$header.$payload" | openssl dgst -sha256 -binary -sign key | openssl enc -base64 | tr -d '\n=' | tr -- '+/' '-_')
+        tokenURL=$(curl -i -X GET -H "Authorization: Bearer $header.$payload.$signature" -H "Accept: application/vnd.github.v3+json" https://api.github.com/app/installations | grep 'access_tokens_url' | awk '{print $2}' | sed -e 's/^"//' -e 's/",$//')
+        token=$(curl -i -X POST -H "Authorization: Bearer $header.$payload.$signature" -H "Accept: application/vnd.github.v3+json" "$tokenURL" | grep 'token'| awk '{print $2}' | sed -e 's/^"//' -e 's/",$//')
+        echo "GITHUB_TOKEN=$token" >> "$GITHUB_ENV"
+    - name: automerge
+      uses: "pascalgn/automerge-action@v0.13.0"
+      env:
+        MERGE_METHOD: "merge"
+        MERGE_LABELS: "automerge,!wip,!do-not-merge/hold"
+        MERGE_RETRY_SLEEP: "300000"
```

Comment:
5 minutes seems a bit too long to retry a merge failure. If the automerge merge process fails, it would be better to know this as soon as possible to avoid further delays. Maybe setting it to 1 min (100000) could be more efficient.

Reply:
@shashankram It's actually set to retry merging every 5 min because it'll keep checking in on when the status checks are finished. And then the e2e tests take on average like ~25min